### ISSUE TITLE
add support for debugging tests

### ISSUE THREAD

**Issue by donalmurtagh** [linux-china/vitest-jetbrains-plugin#16]
In IntelliJ (or Webstorm), it is possible to debug Jest tests by choosing the [debug option](https://jmp.sh/VYEEoz1) from the dropdown menu. This allows you to set breakpoints, step through the code, etc.

Would it be possible to support debugging for Vitest tests?

**Comment by linux-china** [comment]
Yes, I know this problem, and debugging needs more works :)  Now you can debug test by https://github.com/linux-china/vitest-jetbrains-plugin#how-to-debug-vitest-tests

**Comment by simonmarshall** [comment]
hi @linux-china - will there be support for debugging individual tests via your plugin?  
per-test debugging & coverage are such killer features

**Comment by linux-china** [comment]
@simonmarshall  I will take a look. Now I use `RunAnything` to launch Vitest tests instead of `Run Configuration`. Maybe I should choose Run Configuration to launch Vitests, and then you can debug Vitest tests from `Run Configuration`.  

It's not hard to add debug support in WebStorm, just create a Vitest base test run configuration, and copy this configuration to run other tests..

![image](https://user-images.githubusercontent.com/46711/182198756-b32f9afb-1a13-43cf-a7dd-685d266e9913.png)

Maybe I should group Vitest run, watch and debug together, and you can choose which one to run.

**Comment by linux-china** [comment]
@donalmurtagh  @simonmarshall  Debug support implemented with version 0.6.0 

![image](https://user-images.githubusercontent.com/46711/182268062-fd388603-1a31-4337-8398-2ce6fbc271ec.png)

**Comment by simonmarshall** [comment]
awesome @linux-china - thanks!
it assumes our working directory is the project root, so we have to manually adjust each run config to add the appropriate sub-folder, but i guess there's no way around that(?)

**Comment by simonmarshall** [comment]
next request: please support _coverage_ as well as _debug_!  :)

**Comment by linux-china** [comment]
> next request: please support _coverage_ as well as _debug_! :)

Please add `c8` dependency in package.json and reopen the project.  If c8 found in package.json, and Vitest Runner will add `--coverage` automatically.

**Comment by linux-china** [comment]
> awesome @linux-china - thanks! it assumes our working directory is the project root, so we have to manually adjust each run config to add the appropriate sub-folder, but i guess there's no way around that(?)

# How Vitest Runner plugin resolves working directory?
          
* If `workspaces` declared in `package.json`, and working directory is project's root directory.
* Vitest Runner will resolve proximate `package.json` for Vitest tested file, and working directory may be project's root directory or subdirectory of subproject.
                
**Attention**: for mono repository with subprojects independent, working directory will be subproject's directory, 
and Vitest Runner will use `vite.config.js` inside subproject.

**Comment by simonmarshall** [comment]
thanks @linux-china - i'll try both issues when 0.6.1 comes out (we do already have a `c8` dev dep so assumed coverage wasn't supported)

**Comment by simonmarshall** [comment]
> > next request: please support _coverage_ as well as _debug_! :)
> 
> Please add `c8` dependency in package.json and reopen the project. If c8 found in package.json, and Vitest Runner will add `--coverage` automatically. Fixed in version 0.6.1

we do have a `c8` 7.12.0 dev dep in our subproject's package.json, but vitest runner doesn't seem to add `--coverage` to the normal runner if that's what you mean:
```
npm.cmd exec -- vitest run -t "fubar" src/foo/bar.spec.ts

 RUN  v0.20.2 C:/Users/Snafu/Repos/repo/subproject
```

i was assuming it would be available as a `Coverage` entry in the dropdown, like `Watch` & `Debug`, but i don't see it there either.

**Comment by linux-china** [comment]
@simonmarshall  c8 should be declared in root package.json.   Vitest runner will not detect every package.json in sub projects. 

```json
{
  "devDependencies": {
   "c8": "^7.12.0"
  }
}
```

**Comment by simonmarshall** [comment]
> @simonmarshall c8 should be declared in root package.json. Vitest runner will not detect every package.json in sub projects.

Ok, I think I tried that too, without any luck. I'm traveling for a couple of weeks, but will try again when I can.

**Comment by simonmarshall** [comment]
hi @linux-china - i can confirm with 0.6.3 that even if c8 is in devDependencies in the root package.json that there is no `Coverage` entry in the dropdown menu, alongside `Watch` & `Debug`.

**Comment by linux-china** [comment]
Could you try `"@vitest/coverage-c8": "^0.22.1",` for a test?  

![image](https://user-images.githubusercontent.com/46711/186213440-b6bdd259-849f-45c9-9cb6-22673a3f64c3.png)

**Comment by simonmarshall** [comment]
where should i see this happen?  

if it's a normal test run, via green triangle, i can't test that due to the windows path issues
if it's under the dropdown menu, alongside Watch & Debug, then no there's no Coverage entry

i would expect it under the dropdown menu, but maybe my expectations are wrong

**Comment by ewoelfel** [comment]
> awesome @linux-china - thanks! it assumes our working directory is the project root, so we have to manually adjust each run config to add the appropriate sub-folder, but i guess there's no way around that(?)
the pr https://github.com/linux-china/vitest-jetbrains-plugin/pull/35 should fix this behaviour
